server/src/main/java/io/crate/expression/symbol/Function.java

```diff
@@ -364,7 +367,12 @@ private void printCastFunction(StringBuilder builder, Style style) {
         } else {
             asTypeName = " AS " + dataType.getName();
         }
-        builder.append(info.ident().name())
+        var name = info.ident().name();
+        // print cast implicit function under 'cast' name.
+        if (name.equalsIgnoreCase(IMPLICIT_CAST_NAME)) {
```

Comment:
I am not sure about this point, both implicit and explicit cast functions are printed under the same name `cast`. I think it would be better to print implicit cast function under its own name `_cast` and adjust tests. wdyt?

Reply:
We could also consider to omit implicit casts when printing an expression, so that the printed expression is closer to the original input.

Reply:
Some planner would still have to be adjusted to the generic way of printing the implicit cast function. Would it be ok? In most of the rest tests, we can go with `isFunction`.